pkg/remote/terraform/provider.go

```diff
@@ -157,6 +157,10 @@ func (p *TerraformProvider) ReadResource(args tf.ReadResourceArgs) (*cty.Value,
 		delete(args.Attributes, "alias")
 	}
 
+	if alias != p.Config.DefaultAlias {
```

Comment:
This is not the right place to do this, that's not what we want it will break the behavior of the grpcProvider context.

The work should be done in the S3 bucket supplier.

Reply:
We didn't understand each other when I tried to explain it on Discord then.